### PR TITLE
[process-agent] use procutil models for process checks

### PR DESCRIPTION
**Pull request by shang-wang**
### What does this PR do?

We want to move away from gopsutil as much as possible since procutil is in use. This PR tries to remove all references of process model from gopsutil to procutil. Also updated the rt checks to use `Stats` model instead of full process model.

### Motivation

@DataDog/processes 

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Write there any instructions and details that should be tested during the QA.

### CODE REVIEW

File: pkg/process/checks/process.go [hunk 1]
```diff
@@ -15,7 +15,6 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/util/containers"
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 	"github.com/DataDog/gopsutil/cpu"
```
Comment (robertjli) [inline pkg/process/checks/process.go:17]: Should this be removed too, or is it coming later?

File: pkg/process/checks/process_rt.go [hunk 2]
```diff
@@ -119,25 +118,25 @@ func fmtProcessStats(
 	chunked := make([][]*model.ProcessStat, 0)
 	chunk := make([]*model.ProcessStat, 0, cfg.MaxPerMessage)
 
-	for _, fp := range procs {
+	for pid, fp := range procs {
 		// Skipping any processes that didn't exist in the previous run.
 		// This means short-lived processes (<2s) will never be captured.
```
Comment (robertjli) [inline pkg/process/checks/process_rt.go:123]: Nit: this should be `<10s`, right?

File: pkg/process/procutil/process_model.go [hunk 3]
```diff
@@ -187,3 +187,118 @@ func ConvertCtxSwitches(s *NumCtxSwitchesStat) *process.NumCtxSwitchesStat {
...
+func ConvertAllFilledProcesses(processes map[int32]*process.FilledProcess) map[int32]*Process {
+	result := make(map[int32]*Process, len(processes))
+	for pid, p := range processes {
+		result[pid] = ConvertFromFilledProcess(p)
+	}
+	return result
+}
+
+// ConvertFillProcessesToStats takes a group of FilledProcess objects and convert them into Stats
+func ConvertFillProcessesToStats(processes map[int32]*process.FilledProcess) map[int32]*Stats {
```
Comment (robertjli) [inline pkg/process/procutil/process_model.go:201]: Nit: `Fill` -> `Filled`

**Review by robertjli — APPROVED** [review]

File: pkg/process/checks/process_rt.go (hunk 2 above)
Comment (shang-wang) [reply pkg/process/checks/process_rt.go:123]: Ah right.

File: pkg/process/procutil/process_model.go (hunk 3 above)
Comment (shang-wang) [reply pkg/process/procutil/process_model.go:201]: Ah right.

File: pkg/process/checks/process_rt.go (hunk 2 above)
Comment (shang-wang) [reply pkg/process/checks/process_rt.go:123]: Well it is comparing with `lastProcs` which is generated 2s ago. In the context of whole running process-agent you are right, but in the context of rt_process alone it's still 2s.

Comment (robertjli) [reply pkg/process/checks/process_rt.go:123]: Oh yeah, that makes sense. Thanks for explaining!

File: pkg/process/procutil/process_model.go [hunk 4]
```diff
@@ -187,3 +187,118 @@ func ConvertCtxSwitches(s *NumCtxSwitchesStat) *process.NumCtxSwitchesStat {
...
+		Ppid:     p.Ppid,
+		NsPid:    p.NsPid,
+		Name:     p.Name,
+		Cwd:      p.Cwd,
+		Exe:      p.Exe,
+		Cmdline:  p.Cmdline,
+		Username: p.Username,
+		Uids:     p.Uids,
+		Gids:     p.Gids,
+		Stats: &Stats{
```
[the remark this answers is not in the document]
Comment (shang-wang) [reply pkg/process/procutil/process_model.go:233]: Sure, I can extract further.

File: pkg/process/checks/process.go (hunk 1 above)
Comment (shang-wang) [reply pkg/process/checks/process.go:17]: Yea this one is coupled a bit tightly in gopsutil, I need time to move it into procutil.

**Pull request merged**